src/adhara/interop-service/corda-decoder/src/main/java/io/adhara/poc/ledger/MerkleProof.java

```diff
@@ -0,0 +1,47 @@
+package io.adhara.poc.ledger;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class MerkleProof {
+	SecureHash[] proof;
+	byte[] flags;
+	SecureHash[] leaves;
+
+	public void print() {
+		System.out.println("Leaves");
+		for (SecureHash p : leaves) {
+			System.out.println(p.toString());
+		}
+		System.out.println("Proof");
+		for (SecureHash p : proof) {
+			System.out.println(p.toString());
+		}
+		System.out.println("Flags");
+		for (byte b : flags) {
+			System.out.println(b);
+		}
+	}
+
+	public boolean equals(MerkleProof other) {
+		if (proof.length != other.proof.length || flags.length != other.flags.length || leaves.length != other.leaves.length)
```

Comment:
[Arrays,.deepEquals](https://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html#deepEquals-java.lang.Object:A-java.lang.Object:A-) compares array lengths and members:

```java
return Arrays.deepEquals(proof, other.proof)
    && Arrays.deepEquals(flags, other.flags)
    && Arrays.deepEquals(leaves, other.leaves);
```